.github/workflows/provider-couchbase.yml

```diff
@@ -0,0 +1,150 @@
+name: provider-couchbase
+
+on:
+  push:
+    tags:
+      - 'provider-couchbase-v[0-9].[0-9]+.[0-9]+'
+  pull_request:
+    paths:
+      - '.github/workflows/provider-couchbase.yml'
+      - 'providers/couchbase/**'
+
+env:
+  working_directory: providers/couchbase
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  build:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform:
+          - os: linux
+            arch: amd64
+            bin: linux_amd64.bin
+          - os: linux
+            arch: arm64
+            bin: linux_arm64.bin
+          - os: darwin
+            arch: amd64
+            bin: darwin_amd64.bin
+          - os: darwin
+            arch: arm64
+            bin: darwin_arm64.bin
+          - os: windows
+            arch: amd64
+            bin: windows_amd64.exe
+
+    steps:
+      # Checkout the repository with submodules
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      # Set up Go environment
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.24'
+
+      # Install dependencies
+      - name: Install dependencies
+        working-directory: ${{ env.working_directory }}
+        run: go mod tidy
+
+      # Build Go binaries for multiple platforms
+      - name: Build binary ${{ matrix.platform.os }}_${{ matrix.platform.arch }}
+        env:
+          GOOS: ${{ matrix.platform.os }}
+          GOARCH: ${{ matrix.platform.arch }}
+        working-directory: ${{ env.working_directory }}
+        run: |
+          echo "Building for $GOOS/$GOARCH to ${{ matrix.platform.bin }}"
+          GOOS=$GOOS GOARCH=$GOARCH go build -o ${{ matrix.platform.bin }}
```

Comment:
no need to specify $GOOS and $GOARCH here, they'll get picked up from the `env:` block